service/src/main/java/bio/terra/cbas/controllers/ControllerUtils.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.cbas.controllers;
+
+import bio.terra.cbas.dependencies.sam.SamService;
+import bio.terra.cbas.util.BearerTokenFilter;
+import java.util.Optional;
+import javax.servlet.ServletRequest;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ControllerUtils {
+  private static final Logger logger = LoggerFactory.getLogger(ControllerUtils.class);
+  private final ServletRequest servletRequest;
+  private final SamService samService;
+
+  @Autowired
+  ControllerUtils(ServletRequest servletRequest, SamService samService) {
+    this.servletRequest = servletRequest;
+    this.samService = samService;
+  }
+
+  Optional<String> getUserToken() {
+    Object token = servletRequest.getAttribute(BearerTokenFilter.ATTRIBUTE_NAME_TOKEN);
+    return Optional.ofNullable((String) token);
+  }
+
+  Optional<UserStatusInfo> getSamUser() {
+    return getUserToken()
+        .map(
+            token -> {
+              try {
+                return samService.getUserStatusInfo(token);
+              } catch (InterruptedException e) {
+                logger.error(e.getMessage(), e.getCause());
+                Thread.currentThread().interrupt();
+                return null;
```

Comment:
I'm not totally sure how to handle this InterruptedException that comes from Sam...

Reply:
If in doubt (or if different callers might want to handle it differently), feel free to just declare it as thrown by the method and then handle it further down the call stack

Reply:
Found some more info [here](https://github.com/DataBiosphere/terra-common-lib/blob/develop/src/main/java/bio/terra/common/sam/exception/SamExceptionFactory.java#L75). Seems like we should catch and rethrow our own exception for consistency's sake rather than deal with Java's builtin